src/output/help.rs

```diff
@@ -642,24 +644,45 @@ impl<'help, 'app, 'parser, 'writer> Help<'help, 'app, 'parser, 'writer> {
         prefix.to_string() + &spec_vals.join(" ")
     }
 
-    fn write_about(&mut self, new_line: bool) -> io::Result<()> {
+    fn write_about(&mut self, before_new_line: bool, after_new_line: bool) -> io::Result<()> {
         let about = if self.use_long {
             self.parser.app.long_about.or(self.parser.app.about)
         } else {
             self.parser.app.about
         };
         if let Some(output) = about {
+            if before_new_line {
+                self.none("\n")?;
+            }
             self.none(text_wrapper(output, self.term_w))?;
-            if new_line {
+            if after_new_line {
                 self.none("\n")?;
             }
         }
         Ok(())
     }
 
-    fn write_author(&mut self, new_line: bool) -> io::Result<()> {
+    fn write_author(&mut self, before_new_line: bool, after_new_line: bool) -> io::Result<()> {
         if let Some(author) = self.parser.app.author {
+            if before_new_line {
+                self.none("\n")?;
+            }
             self.none(text_wrapper(author, self.term_w))?;
+            if after_new_line {
+                self.none("\n")?;
+            }
+        }
+        Ok(())
+    }
+
+    fn write_version(&mut self, new_line: bool) -> io::Result<()> {
+        let version = if self.use_long {
```

Comment:
Nice catch, adding a regression test would be great.

Reply:
Just added a commit rewriting version/long_version test.